test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+			{
+				for (auto &stmt: *_message->mutable_block()->mutable_statements())
+					if (stmt.has_functioncall())
+					{
+						delete stmt.release_functioncall();
+						stmt.clear_functioncall();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_mediumIP,
+			"Remove function call from function body"
+		);
+	}
+);
+
+// Add dataoffset/datasize
+static YPR<Expression> addDataExpr(
+	[](Expression* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Expression>(
+			[](Expression* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::clearExpr(_message);
+				auto unopdata = new UnaryOpData();
+				unopdata->set_identifier(_rand());
+				unopdata->set_op(
+					YPM::EnumTypeConverter<UnaryOpData_UOpData>{}.enumFromSeed(_rand())
+				);
+				_message->set_allocated_unopdata(unopdata);
+			},
+			_message,
+			_seed,
+			YPM::s_mediumIP,
+			"Mutate expression to dataoffset/size"
+		);
+	}
+);
+
+// Add variable reference inside for-loop body
+static YPR<BoundedForStmt> addVarRefInForBody(
+	[](BoundedForStmt* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<BoundedForStmt>(
+			[](BoundedForStmt* _message, YulRandomNumGenerator& _rand)
+			{
+				auto popStmt = new PopStmt();
+				popStmt->set_allocated_expr(YPM::refExpression(_rand));
+				_message->mutable_for_body()->add_statements()->set_allocated_pop(popStmt);
+			},
+			_message,
+			_seed,
+			YPM::s_mediumIP,
+			"Add variable reference in for loop body"
+		);
+	}
+);
+
+// Mutate expression to a function call
+static YPR<Expression> mutateExprToFuncCall(
+	[](Expression* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Expression>(
+			[](Expression* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::clearExpr(_message);
+				auto functionCall = new FunctionCall();
+				functionCall->set_ret(FunctionCall::SINGLE);
+				functionCall->set_func_index(_rand());
+				_message->set_allocated_func_expr(functionCall);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Mutate expression to function call"
+		);
+	}
+);
+
+// Mutate expression to variable reference
+static YPR<Expression> mutateExprToVarRef(
+	[](Expression* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Expression>(
+			[](Expression* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::clearExpr(_message);
+				_message->set_allocated_varref(YPM::varRef(_rand()));
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Mutate expression to a variable reference"
+		);
+	}
+);
+
+// Add varref to statement
+static YPR<Statement> addVarRefToStmt(
+	[](Statement* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Statement>(
+			[](Statement* _message, YulRandomNumGenerator& _rand)
+			{
+				YPM::addArgs(_message, YPM::refExpression, _rand);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Make statement arguments variable references"
+		);
+	}
+);
+
+// Add varrefs to unset statement arguments recursively
+static YPR<Statement> addVarRefToStmtRec(
```

Comment:
A statement argument (usually an expression) is a recursive data type because an expression can reference another expression.

The recursive mutator ensures that all expressions nested as an argument to a statement are initialized e.g., as variable references.